src/server/auth/server/api_server.go

```diff
@@ -1416,22 +1364,164 @@ func (a *apiServer) RestoreAuthToken(ctx context.Context, req *auth.RestoreAuthT
 		}
 	}
 
-	// Check whether the token hash already exists - we don't want to replace an existing token
-	if _, err := col.NewSTM(ctx, a.env.GetEtcdClient(), func(stm col.STM) error {
-		tokens := a.tokens.ReadWrite(stm)
-		var existing auth.TokenInfo
-		err := tokens.Get(req.Token.HashedToken, &existing)
-		if err == nil {
+	if err := func() error {
+		// Check whether the token hash already exists - we don't want to replace an existing token
+		if _, err := a.lookupAuthTokenInfo(ctx, req.Token.HashedToken); err == nil {
```

Comment:
You can probably eliminate this because of the uniqueness constraint now, right? There's no way to restore an existing token.